### PR TITLE
Fixes #12357: Port sshKeyDistribution to multiversionned technique

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/12357

### CODE REVIEW

File: techniques/system/common/1.0/hooks.st [hunk 1]
```diff
@@ -350,3 +350,302 @@ bundle agent runhook_repoGpgKeyManagementGetKeys(json) {
...
+       # File could not be changed - repaired (or non compliant) - but if repaired, it means it's been fixed
+      "error on file" usebundle     => rudder_common_report(
+          "${technique_name}", "${error[${audit_mode[${bundles}]}]}",
+          "${${bundles}.sshkey_distribution_uuid[${array[${bundles}]}]}", "${component_name}", "${${bundles}.sshkey_distribution_tag[${array[${bundles}]}]}",
+          "The keys for user ${${bundles}.sshkey_distribution_name[${array[${bundles}]}]} could not be flushed"
+        ),
+        ifvarclass => "post_hook_flush_ssh_${${bundles}.sshkey_distribution_name[${array[${bundles}]}]}_error.!post_hook_flush_ssh_${${bundles}.sshkey_distribution_name[${array[${bundles}]}]}_repaired";
+
+
+      # Finallly, a key in audit mode was not compliant. We won't try to flush, as the result is unexpected
```
Comment (amousset) [inline techniques/system/common/1.0/hooks.st]: Finally

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**